pages/api/webhooks/customer-created.ts

```diff
@@ -0,0 +1,43 @@
+import { NextApiRequest, NextApiResponse, NextApiHandler } from "next";
+import { SALEOR_DOMAIN_HEADER } from "../../../constants";
+import Klaviyo from "../../../lib/klaviyo";
+
+const expectedEvent = "customer_created";
+
+const handler: NextApiHandler = async (
+  request: NextApiRequest,
+  response: NextApiResponse
+) => {
+  const saleorDomain = request.headers[SALEOR_DOMAIN_HEADER];
+  if (!saleorDomain) {
+    response
+      .status(400)
+      .json({ success: false, message: "Missing saleor domain token." });
+    return;
+  }
+
+  const receivedEvent = request.headers["saleor-event"]?.toString();
+
+  if (receivedEvent !== expectedEvent) {
+    return response
+      .status(400)
+      .json({ success: false, message: "Invalid event" });
+  }
+
+  const context = request.body;
+  const userEmail = context[0]?.user_email;
+
+  if (!userEmail) {
+    return response
```

Comment:
one more thing - in rest of API handlers we tend to run some methods on `response` object and then `return` with implicit `undefined`. Can we just `return response` here?

I'm aware, that it's working (also from TypeScript compiler perspective). I'm just transparency acknowledging it change our current standard.

Any thoughts @zaiste ?

Reply:
Let's discuss this point shortly in the next daily. There are some peculiarities for API Routes in Next.js in that matter. I would see a wrapper for it, so we leverage the type system and it plays nicely with the current API.